internal/bminventory/inventory.go

```diff
@@ -5157,6 +5145,26 @@ func (b *bareMetalInventory) getBoundCluster(db *gorm.DB, infraEnv *common.Infra
 	return nil, nil
 }
 
+// Checks whether the specified obj can be updated by the user.
+// If not allowed, checks whether it can be read by the user.
+// This is required in order to return an appropriate error for
+// users who are only allowed to read.
+func (b *bareMetalInventory) checkAccessToObj(ctx context.Context, obj interface{}, objType string, objId *strfmt.UUID) error {
+	canWrite, err1 := b.authzHandler.HasAccessTo(ctx, obj, auth.UpdateAction)
+	if !canWrite {
+		canRead, err2 := b.authzHandler.HasAccessTo(ctx, obj, auth.ReadAction)
```

Comment:
Are err1 and err2 set only if canWrite/canRead are false?  What happens if we fail to contact AMS?

Reply:
Yeah, we return false in this case.